### PR TITLE
Support making deals with data directly from IPFS nodes

### PR DESCRIPTION
**Pull request by magik6k**
This PR means that now by setting
```
[Client]
  UseIpfs = true
```
In `~/.lotus/config.toml`, users will be able to make deals with data in their local ipfs nodes, like this:
```
$ ipfs add -r SomeData
QmSomeData
$ ./lotus client deal QmSomeData t03119 0.0000000000001 80000
```

### CODE REVIEW

File: lib/ipfsbstore/ipfsbstore.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	"github.com/ipfs/interface-go-ipfs-core/options"
+	"github.com/ipfs/interface-go-ipfs-core/path"
+)
+
+type IpfsBstore struct {
+	ctx context.Context
+	api iface.CoreAPI
+}
+
+func NewIpfsBstore(ctx context.Context) (*IpfsBstore, error) {
```
Comment (whyrusleeping) [inline lib/ipfsbstore/ipfsbstore.go:26]: Maybe allow people to configure the api endpoint for the daemon?

Comment (magik6k) [reply lib/ipfsbstore/ipfsbstore.go:26]: You can with an env var: https://github.com/ipfs/go-ipfs-http-client/blob/master/api.go#L48

We may want to add a config field for this at some point, but an env var is how you tell ipfs where it's repo lives, so most people with custom location will have it set already

**Comment by mishmosh** [comment]
awesome! @jimmylee can you add to lotus docs?

**Comment by jimmylee** [comment]
Yep! Running a deploy now @mishmosh

**Comment by jimmylee** [comment]
Just spoke with @whyrusleeping and @arajasek, since master will have the latest docs in 10 days we won't need to update the documentation immediately. Meanwhile I recommend for anyone who has the capacity to add the documentation to https://filecoin-docs.netlify.app/how-to/build-interacting-with-the-network/#testnet as per @mishmosh's guidance.

**Pull request merged**